src/anomalib/data/utils/split.py

```diff
@@ -51,89 +51,340 @@ class ValSplitMode(str, Enum):
     SYNTHETIC = "synthetic"
 
 
-def concatenate_datasets(datasets: Sequence["data.AnomalibDataset"]) -> "data.AnomalibDataset":
-    """Concatenate multiple datasets into a single dataset object.
+class SplitMode(str, Enum):
+    """Unified splitting mode for both test and validation subsets.
 
-    Args:
-        datasets (Sequence[AnomalibDataset]): Sequence of at least two datasets.
+    This enum represents the available modes for splitting datasets.
 
-    Returns:
-        AnomalibDataset: Dataset that contains the combined samples of all input datasets.
+    Attributes:
+        SYNTHETIC: Generate synthetic data for splitting.
+        PREDEFINED: Use a pre-defined split from an existing source.
+        AUTO: Automatically determine the best splitting strategy.
+
+    Example:
+        >>> mode = SplitMode.AUTO
+        >>> print(mode)
+        SplitMode.AUTO
+        >>> print(mode.value)
+        'auto'
     """
-    concat_dataset = datasets[0]
-    for dataset in datasets[1:]:
-        concat_dataset += dataset
-    return concat_dataset
+
+    SYNTHETIC = "synthetic"
+    PREDEFINED = "predefined"
+    AUTO = "auto"
```

Comment:
Should we have auto? Currently it will perform differently based on the dataset. Currently, it clones the test data for validation in `auto` mode. Since Kolektor does not have a val set, cloning can also just be `predefined`. But maybe I am confused on the intent of `predefined`